scripts/format_notebooks.py

```diff
@@ -239,15 +258,59 @@ def preprocess_cell(self, cell, resources, cell_index):
             # Write notebook file
             if overwrite_notebook:
                 nb_file_loc = str(file_loc.with_suffix(""))
+            elif check_notebook:
+                _fd, nb_file_loc = tempfile.mkstemp()
```

Comment:
Might be a good idea to close and remove the temp files after they're used, or maybe using [tempfile.TemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile) 🤔 

Another possibility is to use the `io` library to read to memory and compare the values—[nbconvert](https://nbconvert.readthedocs.io/en/latest/api/writers.html#nbconvert.writers.StdoutWriter) has a `nbconvert.writers.StdoutWriter` class.

In either case, the disk IO probably wouldn't affect CI speed with the number of files we have

Reply:
> Might be a good idea to close and remove the temp files after they're used, or maybe using tempfile.TemporaryFile 🤔

I'd decided not to use `tempfile.TemporaryFile` because I thought that it had to be used as a context manager with `with`... but that's of course not true. I'll switch. 👍 

> Another possibility is to use the io library to read to memory and compare the values—nbconvert has a nbconvert.writers.StdoutWriter class.

Unfortunately, as the name suggests, that's only writes to `StdoutWriter`, and I think mutating `sys.stdout` to capture it is worse than using a temporary file.

Reply:
I used `TemporaryDirectory`, because the output writes to a different file (it adds a `.ipynb` extension).